===================================================================
RCS file: /cvsroot/mozilla/layout/xul/base/src/tree/src/ nsTreeBodyFrame.cpp,v
retrieving revision 1.328
retrieving revision 1.329
diff -u -r1.328 -r1.329
--- mozilla/layout/xul/base/src/tree/src/nsTreeBodyFrame.cpp	2007/08/23 15:57:55	1.328
+++ mozilla/layout/xul/base/src/tree/src/nsTreeBodyFrame.cpp	2007/08/27 09:51:23	1.329
@@ -2443,16 +2443,16 @@
       width = 0;
   }
 
-  // If no horz scrolling periphery is present, then just
-  // return the width of the columns
+  // If no horz scrolling periphery is present, then just return our width
   if (width == 0) {
-    width = aParts.mColumnsFrame->GetRect().width;
+    mAdjustWidth = 0;
+    width = mRect.width;
+  } else {
+    // Compute the adjustment to the last column. This varies depending on the
+    // visibility of the columnpicker and the scrollbar.
+    mAdjustWidth = mRect.width - aParts.mColumnsFrame->GetRect().width;
   }
 
-  // Compute the adjustment to the last column. This varies depending on the
-  // visibility of the columnpicker and the scrollbar.
-  mAdjustWidth = mRect.width - aParts.mColumnsFrame->GetRect().width;
-
   return width;
 }
 

